lightning/src/ln/channelmanager.rs

```diff
@@ -4609,6 +4624,15 @@ where
 					if let Some((ref mut shutdown_finish, _)) = shutdown_finish {
 						shutdown_finish.0.take();
 					}
+				} else {
+					let mut pending_events = self.pending_events.lock().unwrap();
```

Comment:
Just curious, would this mean something like pushing the event in `handle_update_monitor_completion` as opposed to here?

And what do you mean by "after broadcast on the outbound channel end"? Is there a hook somewhere that notifies `ChannelManager` of when the counterparty broadcasts the fudning transaction (as opposed to just assuming it's implied when they handle our `funding_signed` message)?

Reply:
> More like in handle_channel_resumption once we broadcast the transaction.

Ohh I see, thanks

> Nope, generally it's quite resource intensive and unsafe to act on mempool actions (except for certain pinning scenarios with anchors). When the counterparty opens the channel toward us, we rely on it being confirmed by them or forget the channel after 2016 blocks (2 weeks) otherwise.

Ah ok cool.